src/ng/rootScope.js

```diff
@@ -1350,6 +1318,27 @@ function $RootScopeProvider() {
 
     return $rootScope;
 
+    function invokeListeners(scope, event, listenerArgs, name) {
+      var listeners = scope.$$listeners[name];
+      if (listeners) {
+        if (listeners.$$index !== undefined) {
+          throw $rootScopeMinErr('inevt', '{0} already $emit/$broadcast-ing on scope ({1})', name, scope.$id);
+        }
```

Comment:
shouldn't the `try...finally` block wrap this `if` block too?
Otherwise the `listeners.$$index` doesn't get cleaned up if the `inevt` error is thrown.

Reply:
Or will the original emit/broadcast catch it in its own `try...finally`?

Reply:
The original should reset it in its own `try...finally`